student/src/main/java/tds/student/services/remote/RemoteOpportunityService.java

```diff
@@ -235,7 +236,43 @@ public void denyApproval(final OpportunityInstance oppInstance) throws ReturnSta
 
   @Override
   public TestConfig startTest(final OpportunityInstance oppInstance, final String testKey, final List<String> formKeys) throws ReturnStatusException {
-    return legacyOpportunityService.startTest(oppInstance, testKey, formKeys);
+    TestConfig testConfig = null;
+    
+    if(isLegacyCallsEnabled) {
+      testConfig = legacyOpportunityService.startTest(oppInstance, testKey, formKeys);
+    }
+  
+    if (!isRemoteExamCallsEnabled) {
+      return testConfig;
+    }
+  
+    
+    /* Note that the formKeys argument can be ignored - it is an unused functionality */
+    Response<ExamConfiguration> response = examRepository.startExam(oppInstance.getExamId());
+  
+    if (response.getError().isPresent()) {
+      ValidationError validationError = response.getError().get();
+      String errorMessage = validationError.getTranslatedMessage().isPresent()
+        ? validationError.getTranslatedMessage().get()
+        : validationError.getMessage();
+    
+      throw new ReturnStatusException(errorMessage);
+    }
+    
+    if (!response.getData().isPresent()) {
+      throw new ReturnStatusException(String.format("Invalid response from the exam service when trying to start exam %s", oppInstance.getExamId()));
+    }
+
+    ExamConfiguration examConfiguration = response.getData().get();
+    
+    /* OpportunityService - Conditional at line 288 */
+    if (!examConfiguration.getStatus().equals(ExamStatusCode.STATUS_STARTED)) {
+      throw new ReturnStatusException("Failed to start the exam.");
+    }
```

Comment:
I guess its up to debate, but this ReturnStatusException actually gets displayed in the UI as an error and I'm not sure we want to include code-level details (which the exam status would be). Would be a prime candidate for error logging though. I can be convinced otherwise though